src/py_obs/osc.py

```diff
@@ -14,16 +21,126 @@ def __str__(self) -> str:
+        data = f"(created): {now}"
+        cmd = ["ssh-keygen", "-Y", "sign", "-f", self.ssh_key_path, "-n", realm, "-q"]
+        proc = subprocess.Popen(
+            cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, encoding="utf-8"
+        )
+        stdout, _ = proc.communicate(data)
+
+        match = re.match(
+            r"\A-----BEGIN SSH SIGNATURE-----\n(.*)\n-----END SSH SIGNATURE-----",
+            stdout,
+            re.S,
+        )
+        if not match:
+            raise RuntimeError("Could not extract SSH signature")
+
+        # headers must not contain newlines; removing them makes no difference in base64 encoded text
+        sig = match.group(1).replace("\n", "")
+
+        auth = f'Signature keyId="{self.login}",algorithm="ssh",headers="(created)",created={now},signature="{sig}"'
+        return auth
+
+
+class CookieJar(http.cookiejar.LWPCookieJar):
+    """
+    A wrapper encapsulating LWPCookieJar for use in aiohttp.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if os.path.isfile(self.filename):
+            try:
+                self.load()
+            except http.cookiejar.LoadError:
+                pass
+
+    def filter_cookies(self, request_url):
+        result = []
+        for cookie in self:
+            if cookie.domain == request_url.host or (
+                cookie.domain.startswith(".")
+                and request_url.host.endswith(cookie.domain)
+            ):
+                result.append((cookie.name, cookie.value))
+        return result
+
+    def update_cookies(self, cookies, response_url):
+        for name, cookie in cookies.items():
+            if cookie["max-age"]:
+                now = int(time.time())
+                expires = now + int(cookie["max-age"])
+            elif cookie["expires"]:
+                expires = cookie["expires"]
+            else:
+                expires = None
+
+            c = http.cookiejar.Cookie(
+                version=cookie["version"] or 0,
+                name=name,
+                value=cookie.value,
+                port=None,
+                port_specified=False,
+                domain=cookie["domain"] or None,
+                domain_specified=True,
+                domain_initial_dot=cookie["domain"].startswith("."),
+                path=cookie["path"] or None,
+                path_specified=True,
+                secure=cookie["secure"] or None,
+                expires=expires,
+                discard=False,
+                comment=cookie["comment"] or None,
+                comment_url=None,
+                rest={},
+            )
+            self.set_cookie(c)
+
+        try:
+            os.makedirs(os.path.dirname(self.filename), mode=0o700)
+        except FileExistsError:
+            pass
+
+        self.save()
+
+
 @dataclasses.dataclass
 class Osc:
     username: str
     password: str
-
     api_url: str = "https://api.opensuse.org/"
+    ssh_key_path: str
```

Comment:
You don't have to specify an SSH key to talk to OBS, only for IBS. Hence this should be optional.
```suggestion
    ssh_key_path: str | None = None
```